i2c_mcux_lpi2c_rtio does not build for MCXN947

<!--
**Notes**
Github Discussions (https://github.com/zephyrproject-rtos/zephyr/discussions)
are available to first verify that the issue is a genuine Zephyr bug and not a
consequence of Zephyr services misuse.

This issue list is only for bugs in the main Zephyr code base
(https://github.com/zephyrproject-rtos/zephyr/). If the bug is for a project
fork (such as NCS) specific feature, please open an issue in the fork project
instead.
-->

**Describe the bug**
<!--
A clear and concise description of what the bug is.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - ...
-->
Building for MCXN947 and I2C_RTIO fails due to this driver not being updated with latest MCUX SDK changes (#70866). As a result, applications using RTIO won't work (e.g: Sensors using Model V2).

**To Reproduce**
<!--
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make
4. See error
-->
Build command:
```
west build -b frdm_mcxn947/mcxn947/cpu0 samples/subsys/shell/shell_module -- -DCONFIG_I2C=y -DCONFIG_I2C_SHELL=y -DCONFIG_I2C_RTIO=y
```

Ouptut:
```
[2/247] Generating include/generated/zephyr/version.h
-- Zephyr version: 4.1.99 (/home/ubieda/zephyrproject/zephyr), build: v4.1.0-2109-g4925e22a6798
[240/247] Building C object zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c_rtio.c.obj
FAILED: zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c_rtio.c.obj 
ccache /home/ubieda/zephyr-sdk-0.17.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCPU_MCXN947VDF_cm33_core0 -DKERNEL -DK_HEAP_MEM_POOL_SIZE=1024 -DLPFLEXCOMM_INIT_NOT_USED_IN_DRIVER=1 -DNDEBUG -DPICOLIBC_LONG_LONG_PRINTF_SCANF -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr -I/home/ubieda/zephyrproject/zephyr/include -I/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated -I/home/ubieda/zephyrproject/zephyr/soc/nxp/mcx -I/home/ubieda/zephyrproject/zephyr/include/zephyr/posix -I/home/ubieda/zephyrproject/zephyr/lib/posix/options/getopt -I/home/ubieda/zephyrproject/zephyr/soc/nxp/mcx/mcxn/. -I/home/ubieda/zephyrproject/zephyr/subsys/rtio -I/home/ubieda/zephyrproject/zephyr/subsys/shell/modules/kernel_service/thread/.. -I/home/ubieda/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/ubieda/zephyrproject/zephyr/modules/cmsis/. -I/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/common -I/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/lpflexcomm -I/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/mcx_spc -I/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/port -I/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/cache/cache64 -I/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/lpflexcomm/lpi2c -I/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/lpflexcomm/lpuart -I/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/devices/MCX/MCXN/MCXN947 -I/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/devices/MCX/MCXN/periph -I/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/devices/MCX/MCXN/MCXN947/drivers -isystem /home/ubieda/zephyrproject/zephyr/lib/libc/common/include -Wshadow -fno-strict-aliasing -Os -imacros /home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m33 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/home/ubieda/zephyr-sdk-0.17.0/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/ubieda/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/ubieda/zephyrproject/zephyr/samples/subsys/shell/shell_module=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/ubieda/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/ubieda/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -mcmse -specs=picolibc.specs -D_POSIX_THREADS -std=c99 -MD -MT zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c_rtio.c.obj -MF zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c_rtio.c.obj.d -o zephyr/drivers/i2c/CMakeFiles/drivers__i2c.dir/i2c_mcux_lpi2c_rtio.c.obj -c /home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c: In function 'mcux_lpi2c_isr':
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:283:39: warning: passing argument 1 of 'LPI2C_MasterTransferHandleIRQ' makes integer from pointer without a cast [-Wint-conversion]
  283 |         LPI2C_MasterTransferHandleIRQ(base, &data->handle);
      |                                       ^~~~
      |                                       |
      |                                       LPI2C_Type *
In file included from /home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:18:
/home/ubieda/zephyrproject/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/lpflexcomm/lpi2c/fsl_lpi2c.h:953:45: note: expected 'uint32_t' {aka 'unsigned int'} but argument is of type 'LPI2C_Type *'
  953 | void LPI2C_MasterTransferHandleIRQ(uint32_t instance, void *lpi2cMasterHandle);
      |                                    ~~~~~~~~~^~~~~~~~
In file included from /home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:34,
                 from /home/ubieda/zephyrproject/zephyr/include/zephyr/irq_multilevel.h:16,
                 from /home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:20,
                 from /home/ubieda/zephyrproject/zephyr/include/zephyr/device.h:12,
                 from /home/ubieda/zephyrproject/zephyr/include/zephyr/drivers/i2c.h:27,
                 from /home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:13:
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c: In function 'mcux_lpi2c_config_func_0':
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:17176:49: error: 'DT_N_S_soc_S_peripheral_50000000_S_flexcomm_94000_S_lpi2c_94800_IRQ_IDX_0_VAL_priority' undeclared (first use in this function); did you mean 'DT_N_S_soc_S_peripheral_50000000_S_flexcomm_94000_S_lpi2c_94800_REG_IDX_0_VAL_ADDRESS'?
17176 | #define DT_N_INST_0_nxp_lpi2c                   DT_N_S_soc_S_peripheral_50000000_S_flexcomm_94000_S_lpi2c_94800
      |                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
  204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4986:9: note: in expansion of macro 'COND_CODE_1'
 4986 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:100:9: note: in expansion of macro 'BUILD_ASSERT'
  100 |         BUILD_ASSERT(((flags_p & IRQ_ZERO_LATENCY) && \
      |         ^~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:123:9: note: in expansion of macro '_CHECK_PRIO'
  123 |         _CHECK_PRIO(priority_p, flags_p) \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/irq.h:49:9: note: in expansion of macro 'ARCH_IRQ_CONNECT'
   49 |         ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p)
      |         ^~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:385:17: note: in expansion of macro 'IRQ_CONNECT'
  385 |                 IRQ_CONNECT(DT_INST_IRQN(n),                            \
      |                 ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2679:9: note: in expansion of macro 'DT_CAT5'
 2679 |         DT_CAT5(node_id, _IRQ_IDX_, idx, _VAL_, cell)
      |         ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4595:9: note: in expansion of macro 'DT_IRQ_BY_IDX'
 4595 |         DT_IRQ_BY_IDX(DT_DRV_INST(inst), idx, cell)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:349:31: note: in expansion of macro 'UTIL_CAT'
  349 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                               ^~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:3909:27: note: in expansion of macro 'DT_INST'
 3909 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
      |                           ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4595:23: note: in expansion of macro 'DT_DRV_INST'
 4595 |         DT_IRQ_BY_IDX(DT_DRV_INST(inst), idx, cell)
      |                       ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4641:33: note: in expansion of macro 'DT_INST_IRQ_BY_IDX'
 4641 | #define DT_INST_IRQ(inst, cell) DT_INST_IRQ_BY_IDX(inst, 0, cell)
      |                                 ^~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:386:33: note: in expansion of macro 'DT_INST_IRQ'
  386 |                                 DT_INST_IRQ(n, priority),               \
      |                                 ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:27489:44: note: in expansion of macro 'I2C_MCUX_LPI2C_INIT'
27489 | #define DT_FOREACH_OKAY_INST_nxp_lpi2c(fn) fn(0) fn(1)
      |                                            ^~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:393:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  393 | DT_INST_FOREACH_STATUS_OKAY(I2C_MCUX_LPI2C_INIT)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:17176:49: note: each undeclared identifier is reported only once for each function it appears in
17176 | #define DT_N_INST_0_nxp_lpi2c                   DT_N_S_soc_S_peripheral_50000000_S_flexcomm_94000_S_lpi2c_94800
      |                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
  204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4986:9: note: in expansion of macro 'COND_CODE_1'
 4986 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:100:9: note: in expansion of macro 'BUILD_ASSERT'
  100 |         BUILD_ASSERT(((flags_p & IRQ_ZERO_LATENCY) && \
      |         ^~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:123:9: note: in expansion of macro '_CHECK_PRIO'
  123 |         _CHECK_PRIO(priority_p, flags_p) \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/irq.h:49:9: note: in expansion of macro 'ARCH_IRQ_CONNECT'
   49 |         ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p)
      |         ^~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:385:17: note: in expansion of macro 'IRQ_CONNECT'
  385 |                 IRQ_CONNECT(DT_INST_IRQN(n),                            \
      |                 ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2679:9: note: in expansion of macro 'DT_CAT5'
 2679 |         DT_CAT5(node_id, _IRQ_IDX_, idx, _VAL_, cell)
      |         ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4595:9: note: in expansion of macro 'DT_IRQ_BY_IDX'
 4595 |         DT_IRQ_BY_IDX(DT_DRV_INST(inst), idx, cell)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:349:31: note: in expansion of macro 'UTIL_CAT'
  349 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                               ^~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:3909:27: note: in expansion of macro 'DT_INST'
 3909 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
      |                           ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4595:23: note: in expansion of macro 'DT_DRV_INST'
 4595 |         DT_IRQ_BY_IDX(DT_DRV_INST(inst), idx, cell)
      |                       ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4641:33: note: in expansion of macro 'DT_INST_IRQ_BY_IDX'
 4641 | #define DT_INST_IRQ(inst, cell) DT_INST_IRQ_BY_IDX(inst, 0, cell)
      |                                 ^~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:386:33: note: in expansion of macro 'DT_INST_IRQ'
  386 |                                 DT_INST_IRQ(n, priority),               \
      |                                 ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:27489:44: note: in expansion of macro 'I2C_MCUX_LPI2C_INIT'
27489 | #define DT_FOREACH_OKAY_INST_nxp_lpi2c(fn) fn(0) fn(1)
      |                                            ^~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:393:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  393 | DT_INST_FOREACH_STATUS_OKAY(I2C_MCUX_LPI2C_INIT)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/toolchain/gcc.h:87:51: error: expression in static assertion is not an integer
   87 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert((EXPR), "" MSG)
      |                                                   ^
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
  204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4986:9: note: in expansion of macro 'COND_CODE_1'
 4986 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:100:9: note: in expansion of macro 'BUILD_ASSERT'
  100 |         BUILD_ASSERT(((flags_p & IRQ_ZERO_LATENCY) && \
      |         ^~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:123:9: note: in expansion of macro '_CHECK_PRIO'
  123 |         _CHECK_PRIO(priority_p, flags_p) \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/irq.h:49:9: note: in expansion of macro 'ARCH_IRQ_CONNECT'
   49 |         ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p)
      |         ^~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:385:17: note: in expansion of macro 'IRQ_CONNECT'
  385 |                 IRQ_CONNECT(DT_INST_IRQN(n),                            \
      |                 ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:27489:44: note: in expansion of macro 'I2C_MCUX_LPI2C_INIT'
27489 | #define DT_FOREACH_OKAY_INST_nxp_lpi2c(fn) fn(0) fn(1)
      |                                            ^~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:393:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  393 | DT_INST_FOREACH_STATUS_OKAY(I2C_MCUX_LPI2C_INIT)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:17176:49: error: 'DT_N_S_soc_S_peripheral_50000000_S_flexcomm_94000_S_lpi2c_94800_IRQ_IDX_0_VAL_irq' undeclared (first use in this function); did you mean 'DT_N_S_soc_S_peripheral_50000000_S_flexcomm_94000_S_lpi2c_94800_REG_IDX_0_VAL_SIZE'?
17176 | #define DT_N_INST_0_nxp_lpi2c                   DT_N_S_soc_S_peripheral_50000000_S_flexcomm_94000_S_lpi2c_94800
      |                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
  204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4986:9: note: in expansion of macro 'COND_CODE_1'
 4986 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:124:9: note: in expansion of macro 'Z_ISR_DECLARE'
  124 |         Z_ISR_DECLARE(irq_p, 0, isr_p, isr_param_p); \
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/irq.h:49:9: note: in expansion of macro 'ARCH_IRQ_CONNECT'
   49 |         ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p)
      |         ^~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:385:17: note: in expansion of macro 'IRQ_CONNECT'
  385 |                 IRQ_CONNECT(DT_INST_IRQN(n),                            \
      |                 ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:69:53: note: in expansion of macro '__DEBRACKET'
   69 | #define __GET_ARG2_DEBRACKET(ignore_this, val, ...) __DEBRACKET val
      |                                                     ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
  204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2880:9: note: in expansion of macro 'COND_CODE_1'
 2880 |         COND_CODE_1(IS_ENABLED(CONFIG_MULTI_LEVEL_INTERRUPTS),                                     \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2679:9: note: in expansion of macro 'DT_CAT5'
 2679 |         DT_CAT5(node_id, _IRQ_IDX_, idx, _VAL_, cell)
      |         ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2882:22: note: in expansion of macro 'DT_IRQ_BY_IDX'
 2882 |                     (DT_IRQ_BY_IDX(node_id, idx, irq)))
      |                      ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2894:26: note: in expansion of macro 'DT_IRQN_BY_IDX'
 2894 | #define DT_IRQN(node_id) DT_IRQN_BY_IDX(node_id, 0)
      |                          ^~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4648:28: note: in expansion of macro 'DT_IRQN'
 4648 | #define DT_INST_IRQN(inst) DT_IRQN(DT_DRV_INST(inst))
      |                            ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:349:31: note: in expansion of macro 'UTIL_CAT'
  349 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                               ^~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:3909:27: note: in expansion of macro 'DT_INST'
 3909 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
      |                           ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4648:36: note: in expansion of macro 'DT_DRV_INST'
 4648 | #define DT_INST_IRQN(inst) DT_IRQN(DT_DRV_INST(inst))
      |                                    ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:385:29: note: in expansion of macro 'DT_INST_IRQN'
  385 |                 IRQ_CONNECT(DT_INST_IRQN(n),                            \
      |                             ^~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:27489:44: note: in expansion of macro 'I2C_MCUX_LPI2C_INIT'
27489 | #define DT_FOREACH_OKAY_INST_nxp_lpi2c(fn) fn(0) fn(1)
      |                                            ^~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:393:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  393 | DT_INST_FOREACH_STATUS_OKAY(I2C_MCUX_LPI2C_INIT)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c: In function 'mcux_lpi2c_config_func_1':
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:25056:41: error: 'DT_N_S_soc_S_peripheral_50000000_S_flexcomm_b7000_S_lpi2c_b7800_IRQ_IDX_0_VAL_priority' undeclared (first use in this function); did you mean 'DT_N_S_soc_S_peripheral_50000000_S_flexcomm_b7000_S_lpi2c_b7800_REG_IDX_0_VAL_ADDRESS'?
25056 | #define DT_N_INST_1_nxp_lpi2c           DT_N_S_soc_S_peripheral_50000000_S_flexcomm_b7000_S_lpi2c_b7800
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
  204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4986:9: note: in expansion of macro 'COND_CODE_1'
 4986 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:100:9: note: in expansion of macro 'BUILD_ASSERT'
  100 |         BUILD_ASSERT(((flags_p & IRQ_ZERO_LATENCY) && \
      |         ^~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:123:9: note: in expansion of macro '_CHECK_PRIO'
  123 |         _CHECK_PRIO(priority_p, flags_p) \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/irq.h:49:9: note: in expansion of macro 'ARCH_IRQ_CONNECT'
   49 |         ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p)
      |         ^~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:385:17: note: in expansion of macro 'IRQ_CONNECT'
  385 |                 IRQ_CONNECT(DT_INST_IRQN(n),                            \
      |                 ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2679:9: note: in expansion of macro 'DT_CAT5'
 2679 |         DT_CAT5(node_id, _IRQ_IDX_, idx, _VAL_, cell)
      |         ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4595:9: note: in expansion of macro 'DT_IRQ_BY_IDX'
 4595 |         DT_IRQ_BY_IDX(DT_DRV_INST(inst), idx, cell)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_1_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:349:31: note: in expansion of macro 'UTIL_CAT'
  349 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                               ^~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:3909:27: note: in expansion of macro 'DT_INST'
 3909 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
      |                           ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4595:23: note: in expansion of macro 'DT_DRV_INST'
 4595 |         DT_IRQ_BY_IDX(DT_DRV_INST(inst), idx, cell)
      |                       ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4641:33: note: in expansion of macro 'DT_INST_IRQ_BY_IDX'
 4641 | #define DT_INST_IRQ(inst, cell) DT_INST_IRQ_BY_IDX(inst, 0, cell)
      |                                 ^~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:386:33: note: in expansion of macro 'DT_INST_IRQ'
  386 |                                 DT_INST_IRQ(n, priority),               \
      |                                 ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:27489:50: note: in expansion of macro 'I2C_MCUX_LPI2C_INIT'
27489 | #define DT_FOREACH_OKAY_INST_nxp_lpi2c(fn) fn(0) fn(1)
      |                                                  ^~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:393:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  393 | DT_INST_FOREACH_STATUS_OKAY(I2C_MCUX_LPI2C_INIT)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/toolchain/gcc.h:87:51: error: expression in static assertion is not an integer
   87 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert((EXPR), "" MSG)
      |                                                   ^
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
  204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4986:9: note: in expansion of macro 'COND_CODE_1'
 4986 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:100:9: note: in expansion of macro 'BUILD_ASSERT'
  100 |         BUILD_ASSERT(((flags_p & IRQ_ZERO_LATENCY) && \
      |         ^~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:123:9: note: in expansion of macro '_CHECK_PRIO'
  123 |         _CHECK_PRIO(priority_p, flags_p) \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/irq.h:49:9: note: in expansion of macro 'ARCH_IRQ_CONNECT'
   49 |         ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p)
      |         ^~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:385:17: note: in expansion of macro 'IRQ_CONNECT'
  385 |                 IRQ_CONNECT(DT_INST_IRQN(n),                            \
      |                 ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:27489:50: note: in expansion of macro 'I2C_MCUX_LPI2C_INIT'
27489 | #define DT_FOREACH_OKAY_INST_nxp_lpi2c(fn) fn(0) fn(1)
      |                                                  ^~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:393:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  393 | DT_INST_FOREACH_STATUS_OKAY(I2C_MCUX_LPI2C_INIT)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:25056:41: error: 'DT_N_S_soc_S_peripheral_50000000_S_flexcomm_b7000_S_lpi2c_b7800_IRQ_IDX_0_VAL_irq' undeclared (first use in this function); did you mean 'DT_N_S_soc_S_peripheral_50000000_S_flexcomm_b7000_S_lpi2c_b7800_REG_IDX_0_VAL_SIZE'?
25056 | #define DT_N_INST_1_nxp_lpi2c           DT_N_S_soc_S_peripheral_50000000_S_flexcomm_b7000_S_lpi2c_b7800
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
  204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4986:9: note: in expansion of macro 'COND_CODE_1'
 4986 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/arch/arm/irq.h:124:9: note: in expansion of macro 'Z_ISR_DECLARE'
  124 |         Z_ISR_DECLARE(irq_p, 0, isr_p, isr_param_p); \
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/irq.h:49:9: note: in expansion of macro 'ARCH_IRQ_CONNECT'
   49 |         ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p)
      |         ^~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:385:17: note: in expansion of macro 'IRQ_CONNECT'
  385 |                 IRQ_CONNECT(DT_INST_IRQN(n),                            \
      |                 ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:69:53: note: in expansion of macro '__DEBRACKET'
   69 | #define __GET_ARG2_DEBRACKET(ignore_this, val, ...) __DEBRACKET val
      |                                                     ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1'
  204 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2880:9: note: in expansion of macro 'COND_CODE_1'
 2880 |         COND_CODE_1(IS_ENABLED(CONFIG_MULTI_LEVEL_INTERRUPTS),                                     \
      |         ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2679:9: note: in expansion of macro 'DT_CAT5'
 2679 |         DT_CAT5(node_id, _IRQ_IDX_, idx, _VAL_, cell)
      |         ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2882:22: note: in expansion of macro 'DT_IRQ_BY_IDX'
 2882 |                     (DT_IRQ_BY_IDX(node_id, idx, irq)))
      |                      ^~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:2894:26: note: in expansion of macro 'DT_IRQN_BY_IDX'
 2894 | #define DT_IRQN(node_id) DT_IRQN_BY_IDX(node_id, 0)
      |                          ^~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4648:28: note: in expansion of macro 'DT_IRQN'
 4648 | #define DT_INST_IRQN(inst) DT_IRQN(DT_DRV_INST(inst))
      |                            ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_1_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:349:31: note: in expansion of macro 'UTIL_CAT'
  349 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                               ^~~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:3909:27: note: in expansion of macro 'DT_INST'
 3909 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
      |                           ^~~~~~~
/home/ubieda/zephyrproject/zephyr/include/zephyr/devicetree.h:4648:36: note: in expansion of macro 'DT_DRV_INST'
 4648 | #define DT_INST_IRQN(inst) DT_IRQN(DT_DRV_INST(inst))
      |                                    ^~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:385:29: note: in expansion of macro 'DT_INST_IRQN'
  385 |                 IRQ_CONNECT(DT_INST_IRQN(n),                            \
      |                             ^~~~~~~~~~~~
/home/ubieda/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:27489:50: note: in expansion of macro 'I2C_MCUX_LPI2C_INIT'
27489 | #define DT_FOREACH_OKAY_INST_nxp_lpi2c(fn) fn(0) fn(1)
      |                                                  ^~
/home/ubieda/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpi2c'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/ubieda/zephyrproject/zephyr/drivers/i2c/i2c_mcux_lpi2c_rtio.c:393:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  393 | DT_INST_FOREACH_STATUS_OKAY(I2C_MCUX_LPI2C_INIT)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

**Expected behavior**
Driver compiles and works.

--- a/drivers/i2c/i2c_mcux_lpi2c_rtio.c
+++ b/drivers/i2c/i2c_mcux_lpi2c_rtio.c
@@ -16,6 +16,9 @@
 #include <zephyr/kernel.h>
 #include <zephyr/irq.h>
 #include <fsl_lpi2c.h>
+#if CONFIG_NXP_LP_FLEXCOMM
+#include <zephyr/drivers/mfd/nxp_lp_flexcomm.h>
+#endif
 
 #include <zephyr/drivers/pinctrl.h>
 
@@ -40,6 +43,9 @@ LOG_MODULE_REGISTER(mcux_lpi2c);
 
 struct mcux_lpi2c_config {
 	DEVICE_MMIO_NAMED_ROM(reg_base);
+#ifdef CONFIG_NXP_LP_FLEXCOMM
+	const struct device *parent_dev;
+#endif
 	const struct device *clock_dev;
 	clock_control_subsys_t clock_subsys;
 	void (*irq_config_func)(const struct device *dev);
@@ -329,7 +335,15 @@ static int mcux_lpi2c_init(const struct device *dev)
 		return error;
 	}
 
+#if CONFIG_NXP_LP_FLEXCOMM
+	/* When using LP Flexcomm driver, register the interrupt handler
+	 * so we receive notification from the LP Flexcomm interrupt handler.
+	 */
+	nxp_lp_flexcomm_setirqhandler(config->parent_dev, dev,
+				      LP_FLEXCOMM_PERIPH_LPI2C, mcux_lpi2c_isr);
+#else
 	config->irq_config_func(dev);
+#endif
 
 	i2c_rtio_init(data->ctx, dev);
 
@@ -363,6 +377,13 @@ static DEVICE_API(i2c, mcux_lpi2c_driver_api) = {
 	IF_ENABLED(DT_INST_IRQ_HAS_IDX(n, 0),				\
 		(I2C_MCUX_LPI2C_MODULE_IRQ_CONNECT(n)))
 
+#ifdef CONFIG_NXP_LP_FLEXCOMM
+#define PARENT_DEV(n)							\
+	.parent_dev = DEVICE_DT_GET(DT_INST_PARENT(n)),
+#else
+#define PARENT_DEV(n)
+#endif /* CONFIG_NXP_LP_FLEXCOMM */
+
 #define I2C_MCUX_LPI2C_INIT(n)						\
 	PINCTRL_DT_INST_DEFINE(n);					\
 									\
@@ -370,6 +391,7 @@ static DEVICE_API(i2c, mcux_lpi2c_driver_api) = {
 									\
 	static const struct mcux_lpi2c_config mcux_lpi2c_config_##n = {	\
 		DEVICE_MMIO_NAMED_ROM_INIT(reg_base, DT_DRV_INST(n)),	\
+		PARENT_DEV(n)						\
 		.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),	\
 		.clock_subsys =						\
 			(clock_control_subsys_t)DT_INST_CLOCKS_CELL(n, name),\

--- a/drivers/i2c/i2c_mcux_lpi2c_rtio.c
+++ b/drivers/i2c/i2c_mcux_lpi2c_rtio.c
@@ -280,7 +280,11 @@ static void mcux_lpi2c_isr(const struct device *dev)
 	struct mcux_lpi2c_data *data = dev->data;
 	LPI2C_Type *base = (LPI2C_Type *)DEVICE_MMIO_NAMED_GET(dev, reg_base);
 
+#if CONFIG_HAS_MCUX_FLEXCOMM
+	LPI2C_MasterTransferHandleIRQ(LPI2C_GetInstance(base), &data->handle);
+#else
 	LPI2C_MasterTransferHandleIRQ(base, &data->handle);
+#endif
 }
 
 static int mcux_lpi2c_init(const struct device *dev)

--- a/drivers/i2c/i2c_mcux_lpi2c_rtio.c
+++ b/drivers/i2c/i2c_mcux_lpi2c_rtio.c
@@ -350,6 +350,19 @@ static DEVICE_API(i2c, mcux_lpi2c_driver_api) = {
 #define I2C_MCUX_LPI2C_SDA_INIT(n)
 #endif /* CONFIG_I2C_MCUX_LPI2C_BUS_RECOVERY */
 
+#define I2C_MCUX_LPI2C_MODULE_IRQ_CONNECT(n)				\
+	do {								\
+		IRQ_CONNECT(DT_INST_IRQN(n),				\
+			DT_INST_IRQ(n, priority),			\
+			mcux_lpi2c_isr,					\
+			DEVICE_DT_INST_GET(n), 0);			\
+		irq_enable(DT_INST_IRQN(n));				\
+	} while (false)
+
+#define I2C_MCUX_LPI2C_MODULE_IRQ(n)					\
+	IF_ENABLED(DT_INST_IRQ_HAS_IDX(n, 0),				\
+		(I2C_MCUX_LPI2C_MODULE_IRQ_CONNECT(n)))
+
 #define I2C_MCUX_LPI2C_INIT(n)						\
 	PINCTRL_DT_INST_DEFINE(n);					\
 									\
@@ -386,12 +399,7 @@ static DEVICE_API(i2c, mcux_lpi2c_driver_api) = {
 									\
 	static void mcux_lpi2c_config_func_##n(const struct device *dev)\
 	{								\
-		IRQ_CONNECT(DT_INST_IRQN(n),				\
-				DT_INST_IRQ(n, priority),		\
-				mcux_lpi2c_isr,				\
-				DEVICE_DT_INST_GET(n), 0);		\
-									\
-		irq_enable(DT_INST_IRQN(n));				\
+		I2C_MCUX_LPI2C_MODULE_IRQ(n);				\
 	}
 
 DT_INST_FOREACH_STATUS_OKAY(I2C_MCUX_LPI2C_INIT)

--- a/drivers/i2c/i2c_mcux_lpi2c_rtio.c
+++ b/drivers/i2c/i2c_mcux_lpi2c_rtio.c
@@ -43,9 +43,6 @@ LOG_MODULE_REGISTER(mcux_lpi2c);
 
 struct mcux_lpi2c_config {
 	DEVICE_MMIO_NAMED_ROM(reg_base);
-#ifdef CONFIG_NXP_LP_FLEXCOMM
-	const struct device *parent_dev;
-#endif
 	const struct device *clock_dev;
 	clock_control_subsys_t clock_subsys;
 	void (*irq_config_func)(const struct device *dev);
@@ -335,15 +332,7 @@ static int mcux_lpi2c_init(const struct device *dev)
 		return error;
 	}
 
-#if CONFIG_NXP_LP_FLEXCOMM
-	/* When using LP Flexcomm driver, register the interrupt handler
-	 * so we receive notification from the LP Flexcomm interrupt handler.
-	 */
-	nxp_lp_flexcomm_setirqhandler(config->parent_dev, dev,
-				      LP_FLEXCOMM_PERIPH_LPI2C, mcux_lpi2c_isr);
-#else
 	config->irq_config_func(dev);
-#endif
 
 	i2c_rtio_init(data->ctx, dev);
 
@@ -377,21 +366,31 @@ static DEVICE_API(i2c, mcux_lpi2c_driver_api) = {
 	IF_ENABLED(DT_INST_IRQ_HAS_IDX(n, 0),				\
 		(I2C_MCUX_LPI2C_MODULE_IRQ_CONNECT(n)))
 
-#ifdef CONFIG_NXP_LP_FLEXCOMM
-#define PARENT_DEV(n)							\
-	.parent_dev = DEVICE_DT_GET(DT_INST_PARENT(n)),
-#else
-#define PARENT_DEV(n)
-#endif /* CONFIG_NXP_LP_FLEXCOMM */
+/* When using LP Flexcomm driver, register the interrupt handler
+ * so we receive notification from the LP Flexcomm interrupt handler.
+ */
+#define I2C_MCUX_LPI2C_LPFLEXCOMM_IRQ_FUNC(n)				\
+	nxp_lp_flexcomm_setirqhandler(DEVICE_DT_GET(DT_INST_PARENT(n)), \
+					DEVICE_DT_INST_GET(n),		\
+					LP_FLEXCOMM_PERIPH_LPI2C,	\
+					mcux_lpi2c_isr)
+
+#define I2C_MCUX_LPI2C_IRQ_SETUP_FUNC(n)				\
+	COND_CODE_1(DT_NODE_HAS_COMPAT(DT_INST_PARENT(n),		\
+					nxp_lp_flexcomm),		\
+		    (I2C_MCUX_LPI2C_LPFLEXCOMM_IRQ_FUNC(n)),		\
+		    (I2C_MCUX_LPI2C_MODULE_IRQ(n)))
 
 #define I2C_MCUX_LPI2C_INIT(n)						\
 	PINCTRL_DT_INST_DEFINE(n);					\
 									\
-	static void mcux_lpi2c_config_func_##n(const struct device *dev); \
+	static void mcux_lpi2c_config_func_##n(const struct device *dev)\
+	{								\
+		I2C_MCUX_LPI2C_IRQ_SETUP_FUNC(n);			\
+	}								\
 									\
 	static const struct mcux_lpi2c_config mcux_lpi2c_config_##n = {	\
 		DEVICE_MMIO_NAMED_ROM_INIT(reg_base, DT_DRV_INST(n)),	\
-		PARENT_DEV(n)						\
 		.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),	\
 		.clock_subsys =						\
 			(clock_control_subsys_t)DT_INST_CLOCKS_CELL(n, name),\
@@ -417,11 +416,6 @@ static DEVICE_API(i2c, mcux_lpi2c_driver_api) = {
 				&mcux_lpi2c_data_##n,			\
 				&mcux_lpi2c_config_##n, POST_KERNEL,	\
 				CONFIG_I2C_INIT_PRIORITY,		\
-				&mcux_lpi2c_driver_api);		\
-									\
-	static void mcux_lpi2c_config_func_##n(const struct device *dev)\
-	{								\
-		I2C_MCUX_LPI2C_MODULE_IRQ(n);				\
-	}
+				&mcux_lpi2c_driver_api);
 
 DT_INST_FOREACH_STATUS_OKAY(I2C_MCUX_LPI2C_INIT)

